share/nisar/defaults/insar.yaml

```
@@ -297,7 +297,7 @@ runconfig:
                     # whose size is strictly smaller than `min_cluster_pixels`.
                     # When `min_cluster_pixels` is 0, this pre-removal step is
                     # skipped entirely.
-                    min_cluster_pixels: 5
+                    min_cluster_pixels: 2
```

How did you choose this threshold?

---

`min_cluster_pixels` determines which clusters are masked out during the ionosphere filter. Before applying the filter, invalid regions are masked from the unwrapped interferogram based on coherence or water, depending on the input. For example, if the invalid region is defined by a coherence threshold of 0.15, then all unwrapped phase pixels have coherence above 0.15. In this case, clusters smaller than min_cluster_pixels will be removed to reduce potential phase jumps.

By default, the ionosphere filter uses “coherence” and “water” as masks. The choice of min_cluster_pixels is related to coherence bias: if high bias is expected, this value should be increased. Since bias is often stronger in low-coherence areas, the parameter is closely tied to coherence level, though accounting for all factors can be tricky. For now, I'm setting this value based on my experience, but may need to change after looking actual interferograms...